src/main/java/slimeknights/tconstruct/library/recipe/tinkerstation/building/ToolBuildingRecipe.java

```diff
@@ -48,13 +56,58 @@ public class ToolBuildingRecipe implements ITinkerStationRecipe {
   @Getter
   protected final IModifiable output;
   protected final int outputCount;
+  @Nullable
+  protected ResourceLocation layoutSlot;
   protected final List<Ingredient> ingredients;
 
+  /** @deprecated Use {@link #ToolBuildingRecipe(ResourceLocation, String, IModifiable, int, ResourceLocation, List)} */
+  @Deprecated
+  public ToolBuildingRecipe(ResourceLocation id, String group, IModifiable output, int outputCount, List<Ingredient> ingredients) {
+    this(id, group, output, outputCount, null, ingredients);
+  }
+
+  /**
+   * Station slot layout ids and tool definition ids are not always the same
+   * layoutSlot specified in the tool recipe JSON is potentially null,
+   * so we should fallback to the tool definition id
+   * @return StationSlotLayout id
```

Comment:
This `@return` is kinda weird. I'd probably just drop this last line.

You could also simplify this description a lot,
> Gets the ID of the station slot layout for displaying this recipe.
> Typically matches the output definition ID, but some tool recipes share a single layout.

The internal nullability does not matter to the caller.